src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -208,109 +209,119 @@ public void render(Graphics g)
-        if( isTargeting && null != currentPath && null != target)
-        {
-          drawDamagePreviewSet(mapGraphics, currentActor, currentPath, target);
-        }
+        // Get the next event animation if one exists.
+        loadNextEventAnimation();
       }
-      else
+    }
+    else if( getCurrentGameMenu() == null )
+    {
+      mapArtist.drawCursor(mapGraphics, currentActor, isTargeting, myGame.getCursorX(), myGame.getCursorY());
+      Unit target = myGame.gameMap.getLocation(myGame.getCursorX(), myGame.getCursorY()).getResident();
+      if( isTargeting && null != currentPath && null != target)
       {
-        menuArtist.drawMenu(mapGraphics, mapViewX, mapViewY);
+        drawDamagePreviewSet(mapGraphics, currentActor, currentPath, target);
       }
-      
-      // When we draw the map, we want to center it if it's smaller than the view dimensions
-      int deltaX = 0, deltaY = 0;
-      if (mapViewWidth > mapImage.getWidth())
-        deltaX = (mapViewWidth - mapImage.getWidth())/2;
-      if (mapViewHeight > mapImage.getHeight())
-        deltaY = (mapViewHeight - mapImage.getHeight())/2;
-
-      int drawWidth  = Math.min(mapViewWidth,  mapImage.getWidth());
-      int drawHeight = Math.min(mapViewHeight, mapImage.getHeight());
-      // Copy the map image into the window's graphics buffer.
-      // First four coords are the dest x,y,x2,y2. Next four are the source coords.      
-      g.drawImage(mapImage, deltaX, deltaY, deltaX + drawWidth, deltaY + drawHeight,
-                            drawX,  drawY,  drawX  + drawWidth, drawY  + drawHeight,
-                            null);
-
-      // Draw the Commander overlay with available funds.
-      drawCommanderOverlay(g);
+    }
+    else
+    {
+      menuArtist.drawMenu(mapGraphics, mapViewX, mapViewY);
+    }
+
+    // When we draw the map, we want to center it if it's smaller than the view dimensions
+    int deltaX = 0, deltaY = 0;
+    if (mapViewWidth > mapImage.getWidth())
+      deltaX = (mapViewWidth - mapImage.getWidth())/2;
+    if (mapViewHeight > mapImage.getHeight())
+      deltaY = (mapViewHeight - mapImage.getHeight())/2;
+
+    // Copy the relevant section of the map image onto a screen-sized image buffer.
+    Dimension dims = SpriteOptions.getScreenDimensions();
+    int drawScale = SpriteOptions.getDrawScale();
+    BufferedImage screenImage = SpriteLibrary.createTransparentSprite(dims.width/drawScale, dims.height/drawScale);
+    Graphics screenGraphics = screenImage.getGraphics();
+
+    int drawWidth  = Math.min(mapViewWidth,  mapImage.getWidth());
+    int drawHeight = Math.min(mapViewHeight, mapImage.getHeight());
+
+    // First four coords are the dest x,y,x2,y2. Next four are the source coords.
+    screenGraphics.drawImage(mapImage, deltaX, deltaY, (deltaX + drawWidth), (deltaY + drawHeight),
+        drawX,  drawY,  drawX  + drawWidth, drawY  + drawHeight, null);
```

Comment:
Would you mind restoring my old formatting? IMO it's a bit more readable

Reply:
Sure